server/poll/transform.go

```diff
@@ -127,6 +136,44 @@ func (p *Poll) makeAdditionalText(localizer *i18n.Localizer, numberOfVotes int)
 	return strings.Join(lines, "\n")
 }
 
+func progressBarStr(progress float64, width float64) string {
+	progress = math.Min(1.0, math.Max(0, progress))
+	wholeWidth := math.Floor(progress * width)
+	line := ""
+	for i := 0.0; i < wholeWidth; i++ {
+		line += "█"
+	}
+	for i := 0.0; i < (width - wholeWidth); i++ {
+		line += "░"
+	}
+
+	return line
+}
+
+func generateProgressBars(votes map[string]int, numberOfVotes int) []string {
+	lines := make([]string, 0)
+
+	keys := make([]string, 0)
+	for n := range votes {
+		keys = append(keys, n)
+	}
+
+	sort.Strings(keys)
```

Comment:
Is this line needed? This seems to be the cause of the wrong order between progress bars and answer buttons.

Reply:
I was unsure if the order was stable or not for this parameter and thus did a sort to ensure that it always stays in the same order. If its always the same order as the buttons and it remains stable than it can be removed for sure.

Reply:
You're right. The order of keys from map is unstable, so without sorting keys the order of progress bar will change with each run... However, buttons is not ordered alphabetically, so it's going to be particularly weird when adding answers.

So what do you think about for-looping `poll.AnswerOption` here instead of creating `votes` in `ToPostActions`?  `poll.AnswerOption` is slice and the order of elements in slice would be preserved, so without the use of map the order of buttons and progress bars would be matched.

I know it is duplicated for-loop for `poll.AnswerOption`, but it would not be a performance issue because the number of answer options would be a few in most cases.